src/shared/machine-image.c

```diff
@@ -359,7 +359,7 @@ static int image_make(
 
                 block_fd = openat(dfd, filename, O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_NOCTTY);
                 if (block_fd < 0)
-                        log_debug_errno(errno, "Failed to open block device %s/%s, ignoring: %m", path, filename);
+                        log_debug_errno(errno, "Failed to open block device %s/%s, ignoring: %m", strempty(path), filename);
```

Comment:
Your patch would print something bogus if path was really null. And looking at the code, this may be null.

Path is either non-null, and this function is called with some fd, or path==NULL and `dfd==AT_FDCWD`.
So what about doing two things:
1. adding an assert at the top of the function: `assert(path || dfd == AT_FDCWD)`.
2. making this print a proper path always, e.g. something like
```C
_cleanup_free_ char *tmp = NULL;
if (!path)
       safe_getcwd(&tmp);
log_debug_errno(errno, "Failed to open block device %s/%s, ignoring: %m", path ?: strnull(tmp), filename);
```

Reply:
Updated as suggested.